### PR TITLE
Bug 1390866 - Implement API endpoint to GET and PUT a provisioner

### PR DESCRIPTION
**Pull request by helfi92**

### CODE REVIEW

File: schemas/provisioner-response.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+$schema:  http://json-schema.org/draft-04/schema#
+title:          "Provisioner Response"
+description: |
+  Response to an attempt to request data from a provisioner.
```
Comment (djmitche) [inline schemas/provisioner-response.yml]: "Response containing information about a provisioner"

File: src/api.js [hunk 2]
```diff
@@ -2016,6 +2016,97 @@ api.declare({
...
+api.declare({
+  method:     'put',
+  route:      '/provisioners/:provisionerId',
+  name:       'declareProvisioner',
+  stability:  API.stability.experimental,
+  scopes:     [
+    [
+      'queue:declare-provisioner:<provisionerId>#<property>',
+    ],
+  ],
```
Comment (djmitche) [inline src/api.js]: needs `deferAuth` too

Comment (helfi92) [reply src/api.js]: I thought I had added it. Good catch!

**Comment by helfi92** [comment]
@djmitche Will merge in ~ half an hour if no complaints :)

**Pull request merged**